hw/xbox/nv2a.c

```diff
@@ -2002,6 +2002,22 @@ static GraphicsObject* lookup_graphics_object(PGRAPHState *s,
     return NULL;
 }
 
+/* 16 bit to [0.0, F16_MAX = 511.9375] */
+static float convert_f16_to_float(uint16_t f16) {
+    if (f16 == 0x0000) { return 0.0; }
+    uint32_t i = (f16 << 11) + 0x3C000000;
+    return *(float*)&i;
+}
+
+/* 24 bit to [0.0, F24_MAX] */
+static float convert_f24_to_float(uint32_t f24) {
+    assert(!(f24 >> 24));
+    f24 &= 0xFFFFFF;
+    if (f24 == 0x000000) { return 0.0; }
```

Comment:
The idea is:

* assert = Make sure the parameter is only 24 bits
* Make it 24 bits even is asserts are disabled
<br>(mhh.. maybe this is redundant.., seeing how passing a > 24 bit parameter could already be considered as a bug)
* If f24 is 0 we have to return 0.0
<br>(It's IS redundant at the moment, but I found this code by first RE'ing the double version of it and then trying to create a float version. To be honest: I'm not sure why I don't have to add some constant to f24 after shifting it. So if this ever fails and we do add a constant we'll be reminded about the special 0x000000 case [which is part of the original conversion in the other direction too])

We might also extend these for negative numbers at some point if the Xbox supports that.
If you figure out what kind of float this is feel free to write a proper version for it.
This is limited to [0.0, MAX] range (as it was RE'd from the CLEAR code) and seems to ignore one bit (not sure if sign bit, 2 complement or something else)